app/api/controllers/requestParsers/validators/Validator.scala

```diff
@@ -34,8 +34,8 @@ trait Validator[A <: RawData] {
       case Nil => List()
       case thisLevel :: remainingLevels =>
         thisLevel(data).flatten match {
```

Comment:
I know this is kind of outside the scope of the PR, but I think this would be clearer as an if-else on "if (thisLevel(data).flatten.nonEmpty)"...
Having nested pattern-matches can end up being quite obtuse.
(Maybe one for a future validator-related PR though, if easier?)

Reply:
yeah maybe worth leaving this for now (with optimism for what is soon to come!)